### DISCUSSION TITLE
Use selected snippet for search

### DISCUSSION THREAD

**Discussion by Mirodin** [orbitalquark/textadept#29] [Q&A] [answered]
Hey guys,

I am quite new to textadept but since I love Lua beyond reason I just had to give it a try ;)

1. From "bigger" IDEs I am used to select a piece of code and hit Ctrl+F which pastes the selection into my search field. How can I duplicate this behaviour in textadept?
2. Also I would like to get all occurrences of the piece inside the search field highlighted. The example from the documentation `textadept.editing.highlight_words = textadept.editing.HIGHLIGHT_CURRENT` highlights every word under my cursor (like "Highlight all" button in Firefox). This gets very annoying since there is a lot of jumping on the screen.

Version: textadept v11.0 from Voidlinux repos

Thanks for any help!

**Comment by orbitalquark** [comment, marked as answer]
1. You can re-bind Ctrl+F to pre-populate the find entry text:

    keys['ctrl+f'] = function()
      local text = not buffer.selection_empty and buffer:get_sel_text() or nil
      ui.find.focus{find_entry_text = text}
    end

2. Try `ui.find.highlight_all_matches`. This triggers on find next/prev, including for incremental find.

**Reply by Mirodin** [reply]
Thank you, that was exactly what I wanted!
